internal/controller/imagerepository_controller.go

```diff
@@ -277,17 +278,31 @@ func (r *ImageRepositoryReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	}
 
 	// Make sure, that image repository name is the same as on creation.
-	// Do it here to avoid webhook creation.
+	// If it isn't set message
 	imageRepositoryName := strings.TrimPrefix(imageRepository.Status.Image.URL, fmt.Sprintf("quay.io/%s/", r.QuayOrganization))
 	if imageRepository.Spec.Image.Name != imageRepositoryName {
-		oldName := imageRepository.Spec.Image.Name
-		imageRepository.Spec.Image.Name = imageRepositoryName
-		if err := r.Client.Update(ctx, imageRepository); err != nil {
-			log.Error(err, "failed to revert image repository name", "OldName", oldName, "ExpectedName", imageRepositoryName, l.Action, l.ActionUpdate)
-			return ctrl.Result{}, err
+		imageNameDiffersMessage := fmt.Sprintf("%s from '%s' to '%s'", imageRepositoryNameChangedMessage, imageRepositoryName, imageRepository.Spec.Image.Name)
```

Comment:
> Image repository name changed after creation from '%s' to '%s'

Which means the name changed in the spec, but the controller ignored that, right? Can we mention that in the message?

Maybe something like this:

> Image repository name changed after creation from '%s' to '%s'. This does not change the name of the the actual image repository in the registry. To do that, delete this ImageRepository object and re-create it with a different repository name.

Reply:
improved message